bundles/core/src/main/java/com/adobe/cq/wcm/core/components/models/Button.java

```diff
@@ -38,12 +40,26 @@ default String getText() {
         throw new UnsupportedOperationException();
     }
 
+    /**
+     * Returns the button link.
+     *
+     * @return the button link
+     * @since com.adobe.cq.wcm.core.components.models 12.10.0
+     */
+    @JsonIgnore  // avoid duplicate URL in JSON, keep old property for backward compatibility
```

Comment:
I would display the new property in the JSON as it might contain some information that was not available before (e.g. the link target).

Reply:
i thought about this as well - this would result in duplicate information in the JSON, the URL would be listed even thrice.

simplified example:
```javascript
{
  "link": "/content/page1.html":
   "buttonLink": {
      "valid": true,
      "url": "/content/page1.html",
      "htmlAttributes": {
          "href": "/content/page1.html",
          "target": "_blank"
      }
   }
}
```

or for an invalid link:
```javascript
{
   "buttonLink": {
      "valid": false
   }
}
```

is this intended?
is JSON expected to be "view-agnostic"? - in this case at least the "htmlAttributes" map should be hidden, and the information of the target attribute as well.

Reply:
you're right. Given that the link is an HTML link and not a generic one, we might want to hide the HTML properties in the JSON. But I would hide it in the Model implementation, not in the Model itself.

Reply:
and to confirm: you want to appear the link URL twice the JSON - one with the old property name (for backwards compatibility), and second with the "link" or "buttonLink" object?
and in case of invalid a link object appears with "valid: false"?

Reply:
@gabrielwalt has some ideas about the JSON rendering: I let him chime in the discussion.